Added smoking related fields in medical_history module

arschat opened the pull request:
### Release notes

For `medical_history.json` schema:
- `smoking_status`
- `smoking_pack_years`
- `years_since_smoking_cessation`
- remove `smoking_history`
 
### Reviews requested

- Need 5 Reviewers to approve because this is a major update

hannes-ucsc reviewed (approved):
There are some non-sensical attribute combinations that you may want to disallow. For example, smoking_pack_years must be 0 if `smoking_status` is `never`, or `years_since_smoking_cessation` should be absent unless `smoking_status` is `former`.

arschat commented:
Thank you for you comments @hannes-ucsc . I updated the descriptions and added an if-else validation for the combinations.
Specifically:
```
if smoking_status == "active":
    years_since_smoking_cessation: max value = 0
else if smoking_status == "never":
    years_since_smoking_cessation: type = null
    smoking_pack_years: max value = 0
```

ESapenaVentura reviewed (approved):
LGTM 👍
